src/trait_group.h

```diff
@@ -0,0 +1,202 @@
+#pragma once
+#ifndef TRAIT_GROUP_H
+#define TRAIT_GROUP_H
+
+#include "json.h"
+#include "mutation.h"
+#include "string.h"
+#include "string_id.h"
+
+typedef std::string Trait_group_tag;
+typedef std::vector<trait_id> Trait_list;
```

Comment:
Should probably be inside the namespace, as it is mostly used inside of that namespace.